anndata/base.py

```diff
@@ -1936,13 +1936,14 @@ def _to_dict_fixed_width_arrays(self):
         obs_rec, uns_obs = df_to_records_fixed_width(self._obs)
         var_rec, uns_var = df_to_records_fixed_width(self._var)
         d = {
-            'X': self._X,
             'obs': obs_rec,
             'var': var_rec,
+            'X': self.X,
```

Comment:
This change affects non-zarr codepaths, so I'm not convinced it's the right thing to do...